src/Func.h

```diff
@@ -172,6 +172,7 @@ class ScriptFunc : public Func
 	{
 public:
 	ScriptFunc(const IDPtr& id);
+	ScriptFunc(const ID* id);
```

Comment:
This was something I noticed in general with these changes: the number of raw pointers all over everywhere. I don't think they all need to be made into IntrusivePtrs (and probably shouldn't), but we should be trying to reduce the number of raw pointers if we can.

Reply:
From Slack: Main issue is that the `Traverse` infrastructure works with const methods and raw pointers. Workaround is to have the caller do a `const_cast` just to construct `IntrusivePtr<zeek::detail::ID>`.

Reply:
Here too I'd appreciate a Style Decision: is `const_cast` better than having a parallel constructor? Is that in general the case for methods that might take either `IntrusivePtr`s or `const X*` pointers?

Reply:
Looking at just this comment without much context, but couple thoughts:

1. Because in this specific instance `id` isn't stored by the ctor, I think this can be solved with a single ctor taking a `const ID& id`.
2. More generally, our mix of pointers can be hard to solve / make consistent, but it might help to come up with some coding standards how/when to use which types of pointers, in particular for parameter passing. Personally, I tend to use `const T&` for non-mutable arguments without passing/creating ownership; `const T*` for mutable arguments without passing/creating ownership; and `SmartPtr<T>` if the receiver becomes a new owner of the argument. Then problems like this can occur only in the latter case; and in that case, one defintily doesn't want to pass a raw pointer anyways.

Reply:
I'm okay with changing to `const ID& id` here ... but it strikes me as maybe only a smidge better, as it still requires a new constructor (which is I believe what drew @awelzel's eye here in the first place), and it's also non-idiomatic for how in the code base `Obj`-derived pointers are dealt with, which is almost always either via an `IntrusivePtr` or bare pointer, rather than a reference.

Reply:
Given that, should I go the `const_cast` route? Leave things as they currently are?

Reply:
Latest push replaces this with a `const_cast`